dbschema/product.esdl

```diff
@@ -0,0 +1,127 @@
+module default {
+  abstract type Product extending Engagement::Child {
+    overloaded engagement: LanguageEngagement;
+
+    scriptureReferences: Scripture::Collection {
```

Comment:
Thinking more, this is actually not here in app code - it's on Producible. We chose here to not extend Producible, and I stand by that.
DirectScriptureProduct - should have this.
DerivativeScriptureProduct - should have this computed from the producible or the override
OtherProduct - it was here via Producible, but really unused.

So really it's just the first two that have it.
I'm pretty sure EdgeQL doesn't like it when one concrete owns the prop and another concrete computes it. So that's something to consider too.

Maybe we leave it here, and then a rewrite to set it on derivative (and trigger for producible update).